bokeh/io.py

```diff
@@ -439,17 +434,13 @@ def _get_save_args(state, filename, resources, title):
 
     return filename, resources, title
 
-def _save_helper(obj, filename, resources, title, validate):
+def _save_helper(obj, state, filename, resources, title, validate):
     remove_after = False
-    if isinstance(obj, LayoutDOM):
-        if obj.document is None:
-            Document().add_root(obj)
-            remove_after = True
-        doc = obj.document
-    elif isinstance(obj, Document):
-        doc = obj
-    else:
-        raise RuntimeError("Unable to save object of type '%s'" % type(obj))
+    doc = state.document
+
+    if obj not in doc.roots:
```

Comment:
I think checking `obj.document is None` is better? That handles the case where the `obj` is in a doc whether the doc is the state.document "curdoc" or not. And if `obj.document is None` then the `obj` is not in `doc.roots` so good to add as a fallback